proposals/0013-expanded-comparison-sampling.md

```diff
@@ -0,0 +1,432 @@
+ for non-comparison samplers:
+
+* `CalculateLevelOfDetailUnclamped(SamplerState S, float<num> Location)`
+  returns the raw LOD value determined implicitly using gradients representing
+  how much the _Location_ changes across neighboring pixels regardless of
+  whether it falls outside the valid range.
+* `CalculateLevelOfDetail(SamplerState S, float<num> Location)`
+  returns the implicitly-determined LOD value clamped to the valid range.
+
+There are no `CalculateLevelOfDetail` variants for comparison samplers.
+
+## Motivation
+
+Without the full complement of comparison sampler operations, techniques that
+ require LOD calculation beyond the implicit calculations are missing the full
+ support for different approaches to LOD calculation.
+In particular, this can affect algorithms involving shadow maps and any case
+ where standard samples have comparison sample counterparts that need to match.
+In addition, this represents a discrepancy with comparable platforms where this
+ deficiency makes porting, emulating, or developing for multiple platforms
+ difficult or impossible.
+
+## Proposed solution
+
+To add full comparison sampling support, HLSL will introduce an optional feature
+ that adds these new comparison sampling builtin methods to all texture objects
+ that currently support the non-comparison methods:
+
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias)`
+  performs a comparison sample using the implicitly-determined LOD with the
+  explicitly-provided _Bias_ value added to that result.
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY)`
+   samples using the LOD calculated using the explicitly-provided
+   _DDX_ and _DDY_ gradients.
+
+Only the simplest overloads of the new and existing builtin methods are
+ described in detail here for the sake of brevity,
+Though not included above, all relevant overloads of `SampleBias` and
+ `SampleGrad` will be reflected in their new `SampleCmpBias` and
+ `SampleCmpGrad` counterparts.
+The parameters not explicitly described here will function exactly as they do
+ when used by the non-comparison builtin methods:
+
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias)`
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias, int<num> Offset)`
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias, int<num> Offset, float Clamp)`
+* `SampleCmpBias(SamplerComparisonState S, float<num> Location, float Bias, int<num> Offset, float Clamp, uint Status)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY, int<num> Offset)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY, int<num> Offset, float Clamp)`
+* `SampleCmpGrad(SamplerState S, float<num> Location, float DDX, float DDY, int<num> Offset, float Clamp, uint Status)`
```

Comment:
`DDX` and `DDY` require dimensionality of texture surface (minus array dim - see other comment regarding the `<num>` used in the prototypes).

Reply:
I've changed them to use template-style type indicators, which are correct and potentially less confusing.